windows/src/engine/keyman/UfrmKeymanMenu.pas

```diff
@@ -681,13 +686,13 @@ procedure TfrmKeymanMenu.DrawLayer;   // I3990
   DoRedraw;
 end;
 
-procedure TfrmKeymanMenu.FitToDesktop;
+procedure TfrmKeymanMenu.FitToDesktop(X, Y: Integer);
 var
   m: TMonitor;
   wr, mr: TRect;
 begin
   wr := Rect(Left,Top,Left+Width,Top+Height);
-  m := Screen.MonitorFromRect(wr, mdNearest);
+  m := Screen.MonitorFromPoint(Point(X, Y), mdNearest);
```

Comment:
It is possible the workarea has other taskbars which impact the calculation of how to fit the menu onto the screen. However I have not encountered this in testing at this time.